collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -2411,28 +2432,21 @@ public void denied() {
                     runOnUiThread(() -> ToastUtils.showLongToast(R.string.savepoint_used));
                 }
 
-                // Set saved answer path
                 if (formController.getInstanceFile() == null) {
-
-                    // Create new answer folder.
-                    String time = new SimpleDateFormat("yyyy-MM-dd_HH-mm-ss",
-                            Locale.ENGLISH).format(Calendar.getInstance().getTime());
-                    String file = formPath.substring(formPath.lastIndexOf('/') + 1,
-                            formPath.lastIndexOf('.'));
-                    String path = Collect.INSTANCES_PATH + File.separator + file + "_"
-                            + time;
-                    if (FileUtils.createFolder(path)) {
-                        File instanceFile = new File(path + File.separator + file + "_" + time + ".xml");
-                        formController.setInstanceFile(instanceFile);
-                    }
-
-                    formController.getAuditEventLogger().logEvent(AuditEvent.AuditEventType.FORM_START, true);
+                    createInstanceDirectory(formController);
+                    identityPromptViewModel.setAuditEventLogger(formController.getAuditEventLogger());
+                    identityPromptViewModel.requiresIdentity().observe(this, requiresIdentity -> {
+                        if (!requiresIdentity) {
+                            startFormEntry(formController, warningMsg);
+                        }
+                    });
                 } else {
                     Intent reqIntent = getIntent();
                     boolean showFirst = reqIntent.getBooleanExtra("start", false);
 
                     if (!showFirst) {
                         if (!allowMovingBackwards) {
+                            // How does this happen?
```

Comment:
How does this happen?

Reply:
In `Admin Settings` -> `Form Entry Settings` we have an option called `Moving backwards`, if you uncheck it you can't navigate back in forms.

Reply:
Ah thanks. I'll WIP this PR until I take care of that path.

Reply:
Fixed!